signac/core/synced_collection.py

```diff
@@ -106,28 +122,62 @@ def _load(self):
         pass
 
     @abstractmethod
-    def _sync(self):
+    def _sync(self, data):
         """Write data to underlying backend."""
         pass
 
+    @abstractmethod
+    def write_to_cache(self, data=None):
+        """Write data to cache."""
+        pass
+
+    @abstractmethod
+    def read_from_cache(self):
+        """Read data from cache."""
+        pass
```

Comment:
I would prefer to see these renamed and made into private methods `_read_from_cache` and `_write_to_cache`. (Those names currently conflict with the class methods, which might be removed based on feedback to other comments.)